platform/features/table/src/controllers/MCTTableController.js

```diff
@@ -295,27 +448,25 @@ define(
          * will be sorted before display.
          */
         MCTTableController.prototype.updateRows = function (newRows) {
-            var displayRows = newRows;
+            //Reset visible rows because new row data available.
             this.$scope.visibleRows = [];
+
             this.$scope.overrideRowPositioning = false;
 
+            //Nothing to show because no columns visible
             if (!this.$scope.displayHeaders) {
                 return;
             }
 
-            if (this.$scope.enableFilter) {
-                displayRows = this.filterRows(displayRows);
-            }
-
-            if (this.$scope.enableSort) {
-                displayRows = this.sortRows(displayRows);
-            }
-            this.$scope.displayRows = displayRows;
+            //Apply filters and sort a copy of the the new rows
+            this.filterAndSort((newRows || []).slice(0));
+            //Resize columns appropriately
             this.resize();
         };
 
         /**
-         * Filter rows.
+         * Applies user defined filters to rows. These filters are based on
```

Comment:
Interesting to know what the filters are based on, but what's the return value of the method?